### ISSUE TITLE
Unable to delete index

### ISSUE THREAD

**Issue by mwillfox** [strapdata/elassandra#301]
**Elassandra version**: 
> elassandra-6.2.3.8-1

**Plugins installed**: 
> N/A

**JVM version**:
> java version "1.8.0_152"
Java(TM) SE Runtime Environment (build 1.8.0_152-b16)
Java HotSpot(TM) 64-Bit Server VM (build 25.152-b16, mixed mode)

**OS version**:
> Linux 4.14.123-111.109.amzn2.x86_64 #1 SMP Mon Jun 10 19:37:57 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

**Description of the problem including expected versus actual behavior**:
When we have a single table with multiple indices some indices hang on deletion. For example, when I send a DELETE to the deletion endpoint for the index a HTTP response is never received. Logs on the nodes in the cluster show the following error:

> 2019-07-16 13:59:19,944 ERROR [elasticsearch[172.30.135.124][masterService#updateTask][T#2]] CassandraDaemon.java:231 uncaughtException Exception in thread Thread[elasticsearch[172.30.135.124][masterService#updateTask][T#2],5,main]
java.lang.AssertionError: null
	at org.elasticsearch.cluster.metadata.MetaDataDeleteIndexService$KeyspaceRemovalInfo.removeUsedTables(MetaDataDeleteIndexService.java:139)
	at org.elasticsearch.cluster.metadata.MetaDataDeleteIndexService.lambda$deleteIndices$2(MetaDataDeleteIndexService.java:212)
	at java.util.HashMap.computeIfPresent(HashMap.java:1156)
	at org.elasticsearch.cluster.metadata.MetaDataDeleteIndexService.deleteIndices(MetaDataDeleteIndexService.java:212)
	at org.elasticsearch.cluster.metadata.MetaDataDeleteIndexService$1.execute(MetaDataDeleteIndexService.java:101)
	at org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:52)
	at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:676)
	at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:300)
	at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:221)
	at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:152)
	at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150)
	at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188)
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:573)
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:244)
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:207)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

Expected behavior is that the specific index would be deleted and the HTTP response would be received:
```
{ "acknowledged": true }
```
Note that this issue seems very similar to #286 but the behavior is slightly different. With this particular bug the index simply fails to delete completely and all nodes still show the index as green.

**Steps to reproduce**:
 1. Create new keyspace and table:
```
CREATE KEYSPACE test
  WITH REPLICATION = {
    'class' : 'NetworkTopologyStrategy',
    'DC1': 2
  }
  AND DURABLE_WRITES = true;`
CREATE TABLE test.log (
  id uuid,
  data text,
  log_timestamp timestamp,
  PRIMARY KEY (id)
) WITH default_time_to_live = 259200 AND
  compaction = {
    'compaction_window_size': '2',
    'compaction_window_unit': 'HOURS',
    'class': 'org.apache.cassandra.db.compaction.TimeWindowCompactionStrategy'
  };
```
 2. Create indices:
```
curl -XPUT -H 'Content-Type: application/json'  localhost:9200/log_2019-05-05 -d '{
  "settings": {
      "keyspace":"test",
      "index.partition_function":"toDayIndex log_{0,date,yyyy-MM-dd} log_timestamp",
      "index.partition_function_class":"org.elassandra.index.MessageFormatPartitionFunction"
  },
  "mappings": {
      "log" : { "discover" : ".*" }
  }
}'
curl -XPUT -H 'Content-Type: application/json'  localhost:9200/log_2019-05-06 -d '{
  "settings": {
      "keyspace":"test",
      "index.partition_function":"toDayIndex log_{0,date,yyyy-MM-dd} log_timestamp",
      "index.partition_function_class":"org.elassandra.index.MessageFormatPartitionFunction"
  },
  "mappings": {
      "log" : { "discover" : ".*" }
  }
}'
```
 3. Try to delete either index:
```
curl -XDELETE localhost:9200/log_2019-05-05
```
The DELETE call will hang and eventually time out. The error above will display in `/var/log/cassandra/system.log`. All indices will still show green.

**Comment by vroyer** [comment]
This was fixed in version 3.2.6.17 by the commit 56c48d20fc4.
Please upgrade to 6.2.3.17.
Thanks.

**Comment by mwillfox** [comment]
@vroyer updating to version 6.2.3.17 definitely fixed the issue. Thanks!
